net/nimble/host/src/ble_gatts.c

```diff
@@ -2080,6 +2080,141 @@ ble_gatts_count_cfg(const struct ble_gatt_svc_def *defs)
     return 0;
 }
 
+static const char* ble_gatt_chr_f_names[] = {
+    "BROADCAST",
+    "READ",
+    "WRITE_NO_RSP",
+    "WRITE",
+    "NOTIFY",
+    "INDICATE",
+    "AUTH_SIGN_WRITE",
+    "RELIABLE_WRITE",
+    "AUX_WRITE",
+    "READ_ENC",
+    "READ_AUTHEN",
+    "READ_AUTHOR",
+    "WRITE_ENC",
+    "WRITE_AUTHEN",
+    "WRITE_AUTHOR",
+    NULL
+};
+
+static const char* ble_gatt_dsc_f_names[] = {
+    "READ",
+    "WRITE",
+    "READ_ENC",
+    "READ_AUTHEN",
+    "READ_AUTHOR",
+    "WRITE_ENC",
+    "WRITE_AUTHEN",
+    "WRITE_AUTHOR",
+    NULL
+};
+
+#define BLE_CHR_FLAGS_STR_LEN 180
+
+static char *
+ble_gatts_flags_to_str(uint16_t flags, char *buf,
+                       const char **names)
+{
+    int bit;
+    bool non_empty = false;
+    size_t length = 0;
+
+    buf[0] = '\0';
+    strcpy(buf, "[");
+    length += 1;
+    for (bit = 0; names[bit]; ++bit) {
+        if (flags & (1 << bit)) {
+            length += strlen(names[bit]);
+            if (length + 1 >= BLE_CHR_FLAGS_STR_LEN) {
+                return buf;
+            }
+            if (non_empty) {
+                strcat(buf, "|");
+                length += 1;
+            }
+            strcat(buf, names[bit]);
+            non_empty = true;
+        }
+    }
+    strcat(buf, "]");
+    return buf;
+}
+
+
+#define STRINGIFY(X) #X
+#define FIELD_NAME_LEN STRINGIFY(12)
+#define FIELD_INDENT STRINGIFY(2)
+
+static void
+ble_gatt_show_local_chr(const struct ble_gatt_svc_def *svc,
```

Comment:
This may verge on nit-picking, but I am not sure about putting code to print the registered resources in this file (ble_gatts.c).  This file is already pretty big and ugly, and this new functionality feels different from what is already there.

I would suggest one of the following:

1. 
* Move ble_gatt_show_local_chr() to a new file.
* Create ble_gatt_show_local_svc() in the new file.
* Modify ble_gatts_show_local() such that it just iterates the list of services, calling ble_gatt_show_local_svc() on each one.

This removes most of the new code from ble_gatts.c

2. Add accessor functions for the service list; move all new code into a new file.  ble_gatts_show_local() calls the new accessor function and prints each service.

3. Add a generic ble_gatts_lcl_svc_foreach() function to ble_gatts.c.  This function takes a function pointer which gets called for each registered service.  The new print code could go in a new file and be implemented with a call to the new foreach function.